shared/client/src/state/cooldown.rs

```diff
@@ -189,6 +207,26 @@ impl CooldownStepMetadata {
                         })
                         .map_err(|_| CheckpointError::SendCheckpoint)?;
 
+                    // we put the cleanup step at the end, so that if keep_steps == Some(0) the logic will still work
+                    // we'll just delete the dir after we've uploaded it
+                    if let Some(keep_steps) = keep_steps {
+                        // if we fail in any of the above steps we may wind up not queueing this dir for delete
+                        // but that's probably better than risking having the dir deleted from under us
+                        // for a relatively low priority disk cleanup task
+                        let mut delete_queue_guard = delete_queue.lock().await;
+                        delete_queue_guard.push(Reverse(step));
+                        // in the happy case this could be an if but if previous iterations failed somewhere
+                        // then we may have more than 1 dir to clean up
+                        while delete_queue_guard.len() > keep_steps as usize {
+                            let delete_step = delete_queue_guard.pop().unwrap().0;
+                            let delete_path =
+                                checkpoint_dir.join(format!("{run_id}-step{delete_step}"));
+                            if let Err(err) = tokio::fs::remove_dir_all(delete_path.clone()).await {
+                                warn!("Error removing {} : {}", delete_path.display(), err);
+                            }
+                        }
+                    }
```

Comment:
does this mean that local checkpoints won't be deleted if we don't set `keep_steps` explicitly? I think it would make sense to leave some reasonable default value for `keep_steps` so that disk doesn't fill up in case the user forgets to set this argument